backend/vouch/tests/test_trust_algo.py

```diff
@@ -0,0 +1,125 @@
+
+            if trust_stat[i] > 0:
+                assert rescaled[i] >= 0.999999999
+
+    def setUp(self) -> None:
+        self.user_0 = UserFactory(username=self._user_0)
+        # UserFactory(username=self._user_1)
+        self.user_1 = User.objects.create_user(username=self._user_1, email="user1@trusted.test")
+        self.user_2 = UserFactory(username=self._user_2)
+        # UserFactory(username=self._user_3) #UserFactory(username=self._user_3)
+        self.user_3 = User.objects.create_user(username=self._user_3, email="user3@trusted.test")
+        self.user_4 = UserFactory(username=self._user_4)
+        self.user_5 = UserFactory(username=self._user_5)
+        # UserFactory(username=self._user_6)
+        self.user_6 = User.objects.create_user(username=self._user_6, email="user6@trusted.test")
+        self.user_7 = UserFactory(username=self._user_7)
+        self.user_8 = UserFactory(username=self._user_8)
+        self.user_9 = UserFactory(username=self._user_9)
+
+        email_domain = EmailDomain.objects.filter(domain="@trusted.test").first()
+        email_domain.status = EmailDomain.STATUS_ACCEPTED
+        email_domain.save()
+
+        Voucher.objects.bulk_create(
+            [
+                # user_0 has given zero voucher
+                # user_1 has given three vouchers
+                Voucher(by=self.user_1, to=self.user_0, trust_value=10),
+                Voucher(by=self.user_1, to=self.user_3, trust_value=13),
+                Voucher(by=self.user_1, to=self.user_7, trust_value=17),
+                # user_2 has given one voucher
+                Voucher(by=self.user_2, to=self.user_5, trust_value=15),
+                # user_3 has given two vouchers
+                Voucher(by=self.user_3, to=self.user_1, trust_value=31),
+                Voucher(by=self.user_3, to=self.user_5, trust_value=35),
+                # user_4 has given one voucher
+                Voucher(by=self.user_4, to=self.user_7, trust_value=47),
+                # user_5 has given one voucher
+                Voucher(by=self.user_5, to=self.user_1, trust_value=51),
+                # user_6 has given zero voucher
+                # user_7 has given two vouchers
+                Voucher(by=self.user_7, to=self.user_1, trust_value=71),
+                Voucher(by=self.user_7, to=self.user_2, trust_value=72),
+                # user_8 has given one voucher
+                Voucher(by=self.user_8, to=self.user_3, trust_value=83),
+                # user_9 has given two vouchers
+                Voucher(by=self.user_9, to=self.user_4, trust_value=94),
+                Voucher(by=self.user_9, to=self.user_5, trust_value=95),
+            ]
+        )
+
+    def test_trust_algo(self):
+        trust_algo()
+        users = list(user.User.objects.all())
+        assert users[1].trust_score >= 0.9999999
+        assert users[9].trust_score == pytest.approx(0, 0.00000001)
```

Comment:
These assertions depend only on the email trust status and not on the Vouchers themselves.  
It would be useful to add a check about one of the values that are expected to be modified thanks to the vouchers. Otherwise, a regression in the algorithm could easily go unnoticed.

Reply:
Hi, 
Thank you, I didn't notice that we could obtain the same result with an unworking vouching mechanism. I fixed that and added verification that we can update the vouchers.